### PR TITLE
gitflow-feature-stash: login-api-endpoint

### PR DESCRIPTION
**Pull request by tamirse**

### CODE REVIEW

File: config/default.js [hunk 1]
```diff
@@ -3,35 +3,41 @@ module.exports = {
...
+        clientId:
+          '409833050808-omel02bdcf5h2pvbmc7k1mg8l8okh9u0.apps.googleusercontent.com',
+      },
     },
-    port: 3000
   },
   mongo: {
     debug: true,
-    uri: 'mongodb://localhost/gaming',
+    uri: 'mongodb://localhost/admin',
```
Comment (tamirse) [inline config/default.js:18]: some parts of the code should use the gaming route, fix

Comment (leonprou) [reply config/default.js:18]: I wanted to change it to `mongodb://localhost/genie`, why admin?

Comment (tamirse) [reply config/default.js:18]: only because admin already existed. will change to genie

Comment (tamirse) [reply config/default.js:18]: not sure I understand your comment

Comment (leonprou) [reply config/default.js:18]: Sorry did understand your first comment too..
In 'protocol://localhost/name' name is the db name the project will use. If the db doesn't exists on the server in creates one. So imo the db name should be "genie" 
Btw admin i think is the db mongo uses to manage access to the databases and the db server

Comment (tamirse) [reply config/default.js:18]: Ok, now I understand what you meant.
The db with the name "Admin" already existed and I wrongefully assumed you created it.
I've changed it to genie then.

File: src/routes/api/login.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+const router = require('express').Router()
+const jwt = require('jsonwebtoken')
+const config = require('config')
+const mongoose = require('mongoose')
```
Comment (leonprou) [inline src/routes/api/login.js:4]: you should require mongoose from services dir, then you don't need to require the models like in line 6

File: src/models/User.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    },
+    name: { type: String, required: [true, "can't be blank"] },
+    externalId: { type: String, required: [true, "can't be blank"] },
+    provider: { type: String, required: [true, "can't be blank"] },
+    accountAddress: { type: String }
+  },
+  { timestamps: true }
+)
+
+UserSchema.index({ externalId: 1 }, { unique: true })
```
Comment (leonprou) [inline src/models/User.js:20]: - you dont need the externalId. it was meant to connect genie with other dbs (for prices).  Here I don't really care what's the user id on google
- but email should be unique :)

File: package.json [hunk 4]
```diff
@@ -67,4 +68,4 @@
     "nodemon": "^1.18.11",
     "standard": "^12.0.1"
   }
-}
+}
```
Comment (leonprou) [inline package.json:71]: please add a new line to the end :)

File: src/routes/api/login.js (hunk 2 above)
Comment (tamirse) [reply src/routes/api/login.js:4]: then why are you doing this in the Prizes/Funding etc endpoints?

Comment (leonprou) [reply src/routes/api/login.js:4]: In that case you just don't need to resolve `require('../../models') yourself. Sorry didn't explained that. Bur nvm, doesn't really matters

Comment (tamirse) [reply src/routes/api/login.js:4]: Still not sure I understand your comment...
What case are you talking about specifically, and why do I not need to resolve it myself?

**Pull request merged**